packages/leancode_hooks/lib/src/use_disposable.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+
+/// Creates and memoizes an instance created using [builder].
+/// On unmount, executes [dispose] in order to cleanup resources.
+T useDisposable<T>({
+  required ValueGetter<T> builder,
+  required void Function(T) dispose,
+  List<Object?> keys = const <Object>[],
+}) {
+  final value = useMemoized(builder, keys);
+
+  useEffect(
+    () => () => dispose(value),
+    [value],
```

Comment:
Shouldn't this use the same keys as `useMemoized`, so that it's executed the same number of times?

Reply:
It wouldn't really change anything, would it? If those keys change, then the instance returned from `useMemoized` would also be different. 

Technically, you can do stuff like extending `ValueNotifier` and override `hashCode` and the equality operator, but then it's your own fault lol

Reply:
I think I will update it as you mentioned.

Reply:
I mean, the object can be something different entirely (e.g. a complex data object) that happens to have the same value across recalculations